app/src/main/java/com/x8bit/bitwarden/data/auth/repository/util/AuthDiskSourceExtensions.kt

```diff
@@ -183,8 +184,46 @@ val AuthDiskSource.onboardingStatusChangesFlow: Flow<OnboardingStatus?>
         }
         .distinctUntilChanged()
 
+/**
+ * Returns the current [OnboardingStatus] of the active user.
+ */
 val AuthDiskSource.currentOnboardingStatus: OnboardingStatus?
     get() = this
         .userState
         ?.activeUserId
         ?.let { this.getOnboardingStatus(userId = it) }
+
+/**
+ * Returns a [Flow] that emits every time the active user's first time state is changed.
+ */
+@OptIn(ExperimentalCoroutinesApi::class)
+val AuthDiskSource.firstTimeStateFlow: Flow<UserState.FirstTimeState>
+    get() = activeUserIdChangesFlow
+        .flatMapLatest { activeUserId ->
+            combine(
+                listOf(
+                    activeUserId
+                        ?.let {
+                            getShowImportLoginsFlow(it)
+                        }
+                        ?: flowOf(null),
+                ),
+            ) {
+                UserState.FirstTimeState(
+                    showImportLoginsCoachMarker = it[0],
+                )
+            }
+        }
+        .distinctUntilChanged()
+
+val AuthDiskSource.currentOrDefaultUserFirstTimeState
```

Comment:
Docs and tests?

Reply:
<img src="https://media0.giphy.com/media/xT5LMzIK1AdZJ4cYW4/giphy.gif"/>